### PR TITLE
Fixes #36508 - Fix virtual nic configuration in preseed

### PR DESCRIPTION
**Pull request by nadjaheitmann**
Based on https://github.com/theforeman/community-templates/pull/688 
cc @m-bucher

### CODE REVIEW

**Review by sbernhard — COMMENTED** [review]
LGTM

**Review by ekohl — CHANGES_REQUESTED** [review]
I think this is mostly correct, but I still think in https://github.com/theforeman/community-templates/pull/688#issuecomment-653448088 a valid point is raised.

We have on line 48 some code that looks up the real interface name (based on the mac address). Is that not something we should still do? Or at least take into account?

The reason we have that is to rely less on the actual identifier and more on the mac address to identify an interface. With the udev stable interface names they may be different from what users expect, or they may not care as much. Whether this assertion still holds is another, but I fear we may end up with a mismatch if we only partially implement it.

I can also see the logic should be something like:
```erb
<% if interface.virtual? -%>
real=`ip -o link | awk '/<%= interface.attached_to.mac -%>/ {print $2;}' | sed s/:$//`
<% else -%>
real=`ip -o link | awk '/<%= interface.mac -%>/ {print $2;}' | sed s/:$//`
<% end -%>
```

Now this probably won't work because attached_to isn't a foreign key, but I hope it shows the concept.

And if anything else, I'd prefer this to be written at least more minimal to show the flow:
```erb
<% if interface.virtual? -%>
real="<%= interface.attached_to -%>"
<% else -%>
real=`ip -o link | awk '/<%= interface.mac -%>/ {print $2;}' | sed s/:$//`
<% end -%>
```

**Review by ekohl — APPROVED** [review]
Untested, but looks good when I read the docs.

File: app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb [hunk 1]
```diff
@@ -45,10 +45,11 @@ EOF
 <% interface_subnet6 = interface.subnet6 -%>
 <% interface_dhcp6 = interface_subnet6.nil? ? true : interface_subnet6.dhcp_boot_mode? -%>
 <% next if !interface.managed? || (interface_subnet.nil? && interface_subnet6.nil?) || interface.primary -%>
-real=`ip -o link | awk '/<%= interface.mac -%>/ {print $2;}' | sed s/:$//`
 <% virtual = interface.virtual? -%>
```
Comment (nadjaheitmann) [inline app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb:48]: @ekohl I think I should have removed `<% virtual = interface.virtual? -%>` as well, not sure why it is still there. Shall I create a fixup?

Comment (ekohl) [reply app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb:48]: Oh yes, it could be removed now. Good catch.

**Pull request merged**